reconcile/ocm_subscription_labels/integration.py

```diff
@@ -0,0 +1,281 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import (
+    Callable,
+    Iterable,
+)
+from typing import (
+    Any,
+    Optional,
+)
+
+from pydantic import (
+    BaseModel,
+    validator,
+)
+
+from reconcile.gql_definitions.fragments.ocm_environment import OCMEnvironment
+from reconcile.gql_definitions.ocm_subscription_labels.clusters import ClusterV1
+from reconcile.gql_definitions.ocm_subscription_labels.clusters import (
+    query as cluster_query,
+)
+from reconcile.utils import gql
+from reconcile.utils.disabled_integrations import integration_is_enabled
+from reconcile.utils.helpers import flatten
+from reconcile.utils.ocm.clusters import (
+    ClusterDetails,
+    discover_clusters_for_organizations,
+)
+from reconcile.utils.ocm.labels import (
+    add_subscription_labels,
+    delete_subscription_labels,
+    update_subscription_labels,
+)
+from reconcile.utils.ocm_base_client import (
+    OCMAPIClientConfigurationProtocol,
+    OCMBaseClient,
+    init_ocm_base_client,
+)
+from reconcile.utils.runtime.integration import (
+    PydanticRunParams,
+    QontractReconcileIntegration,
+)
+from reconcile.utils.secret_reader import SecretReaderBase
+
+QONTRACT_INTEGRATION = "ocm-subscription-labels"
+
+
+class EnvWithClusters(BaseModel):
+    env: OCMEnvironment
+    clusters: list[ClusterV1] = []
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class ClusterState(BaseModel):
```

Comment:
can we name this `ClusterLabelState`?
the way the `eq` method is implemented is easier to anticipate when the class name indicates that this is about the labels of a cluster

Reply:
changed in [c2d6efc](https://github.com/app-sre/qontract-reconcile/pull/3714/commits/c2d6efc3dbb046cd66f0c242bdb7a887ebd9f3b1)